array.h

```diff
@@ -56,6 +69,30 @@
 // functions have internal linkage.
 #define NDARRAY_UNIQUE static
 
+// Functions attributed with NDARRAY_HOST_DEVICE can run on both device and host mode on CUDA.
+#if defined(__clang__) && defined(__CUDA__) && !defined(__CUDA_ARCH__)
+// clang compiling CUDA code, host mode.
+#define NDARRAY_HOST_DEVICE
```

Comment:
Why do we need this? Or alternatively, if we do this, do we need `__host__` in the `__CUDA_ARCH__` case? But I'd rather just do `__host__ __device__` always for `__CUDA__` as that seems like "standard" CUDA style.

Reply:
You're right - we don't need it to be that specific. Done.